src/bindings/jna/libelektra4j/src/main/java/org/libelektra/exception/LogicalException.java

```diff
@@ -0,0 +1,16 @@
+package org.libelektra.exception;
+
+import org.libelektra.Key;
+
+public abstract class LogicalException extends PermanentException {
+
+	private static String errorCode = "C01300";
+
+	public LogicalException(Key k) {
+		super(k);
+	}
+
+	public static String errorCode() {
```

Comment:
Why no @Override here?

Reply:
Because it is a static method?

Reply:
Exactly, wouldn't that possibly return wrong errorCodes in catch blocks?

Reply:
Why should it? The error code can be returned via `e.getErrorNumber()` which is in the KDBException and actually calls `errorKey.getMeta("error/number").getString()`. The static method is never used in a catch block

Reply:
It would be far more efficient if you improve the docu/code and avoid starting discussions. (Do you really want me to explain why someone might invoke a public method?)

Anyway: If a user should not use the method, why is it there, then? Docu does not give a glue. Furthermore, we again have a mess in code/number.

Reply:
It would not even come into my mind that something like this should be documented because it is a standard java coding style.

```java
try {
//code that causes exception
} catch (KDBException e) { \\or any subexception
	e.getErrorNumber();
}
```
Why should somebody call a static method here?

The static method is just useful so one does not have to provide a no-arg constructor (which would even be misleading and not even be possible for the abstract classes) and use `new XYException().getErrorNumber()` or lookup the error number in the design decisions.